From a1c1aec8bfffb868a7ef650329c2c363f82ac5e7 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sat, 12 Feb 2022 23:37:47 +0000
Subject: [PATCH 6/7] fetcher: fix overriding the max retry count in config
 file

Needed to actually handle str to int conversion and vice versa for the config file.
---
 pisi/fetcher.py | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/pisi/fetcher.py b/pisi/fetcher.py
index 1fed1bc..fb03bb8 100644
--- a/pisi/fetcher.py
+++ b/pisi/fetcher.py
@@ -231,11 +231,15 @@ class Fetcher:
             return 0
 
     def _get_retry_attempts(self):
-        retry_attempts = ctx.config.options.retry_attempts or ctx.config.values.general.retry_attempts
-        if retry_attempts and retry_attempts != "5":
+        retry_attempts = 5
+        if ctx.config.options.retry_attempts != retry_attempts:
+            retry_attempts = ctx.config.options.retry_attempts
+            return retry_attempts
+        elif ctx.config.values.general.retry_attempts != str(retry_attempts):
+            retry_attempts = ctx.config.values.general.retry_attempts
             return int(retry_attempts)
         else:
-            return 5
+            return retry_attempts
 
     def _test_range_support(self):
         if not os.path.exists(self.partial_file):
-- 
2.35.1

